components/event-gateway/internal/events/bus/send.go

```diff
@@ -0,0 +1,97 @@
+package bus
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/kyma-project/kyma/components/event-gateway/internal/events/api"
+	"github.com/kyma-project/kyma/components/event-gateway/internal/httpconsts"
+	"github.com/kyma-project/kyma/components/event-gateway/internal/httptools"
+)
+
+var (
+	httpClientProvider  = httptools.DefaultHttpClientProvider
+	httpRequestProvider = httptools.DefaultHttpRequestProvider
+)
+
+func InitEventSender(clientProvider httptools.HttpClientProvider, requestProvider httptools.HttpRequestProvider) {
+	httpClientProvider = clientProvider
+	httpRequestProvider = requestProvider
+}
+
+// SendEvent sends the incoming request to the Sender
+func SendEvent(req *api.SendEventParameters, traceHeaders *map[string]string) (*api.SendEventResponse, error) {
```

Comment:
I would propose to refactor this function a bit. I would extract the following operations into separate functions : creating http request,  unmarshaling correct response, unmarshalling error response.  Such a change would make SentEvent shorter and easier to read. What do you think?

Reply:
This is a code copy from the `gateway` component.